registry/server/sharedLibs/routes/getSharedLibs.ts

```diff
@@ -5,11 +5,26 @@ import preProcessResponse from '../../common/services/preProcessResponse';
 import { Tables } from '../../db/structure';
 import { appendDigest } from '../../util/hmac';
 import { EntityTypes } from '../../versioning/interfaces';
+import Joi from 'joi';
+import { filtersMiddleware, RequestWithFilters } from '../../middleware/filters';
 
-const getSharedLibs = async (req: Request, res: Response): Promise<void> => {
+interface Filters {
+    name?: string;
+}
+
+const filtersSchema = Joi.object<Filters>({
+    name: Joi.string().optional(),
+});
+
+const getSharedLibs = async (req: RequestWithFilters<Filters>, res: Response): Promise<void> => {
     const query = db
         .selectVersionedRows(Tables.SharedLibs, 'name', EntityTypes.shared_libs, [`${Tables.SharedLibs}.*`])
         .from(Tables.SharedLibs);
+
+    if (req.filters?.name) {
```

Comment:
Could you please add a test involving these queries (if it's not exits yet).

Reply:
✅